bip-0364.mediawiki

```diff
@@ -0,0 +1,197 @@
+<pre>
+  BIP: 364
+  Layer: Consensus (soft fork)
+  Title: 64 bit arithmetic operations
+  Author: Chris Stewart <stewart.chris1234@gmail.com>
+  Comments-Summary: No comments yet.
+  Comments-URI: https://github.com/bitcoin/bips/wiki/Comments:BIP-0364
+  Status: Draft
+  Type: Standards Track
+  Created: 2023-09-11
+  License: BSD-3-Clause
+</pre>
+
+==Abstract==
+
+This BIP describes a new set of opcodes (OP_ADD64, OP_SUB64, OP_MUL64, OP_DIV64, OP_NEG64, 
+OP_LESSTHAN64, OP_LESSTHANOREQUAL64, OP_GREATERTHAN64, OP_GREATERTHANOREQUAL64)
+that allows 64 bit signed integer math in the bitcoin protocol.
+
+==Summary==
+
+The arithmetic opcodes (OP_ADD64, OP_SUB64, OP_MUL64, OP_DIV64) behave as follows 
+
+* Fail if less than 2 elements on the stack 
+* Fail if the stacks top 2 elements are not exactly 8 bytes 
+* If the operation results in an overflow, push false onto the stack 
+* If the operation succeeds without overflow, push the result and true onto the stack 
+* the nSequence field of the txin is 0xffffffff;
+
+64 bit comparison opcodes (OP_LESSTHAN64, OP_LESSTHANOREQUAL64, OP_GREATERTHAN64, OP_GREATERTHANOREQUAL64)
+* Fail if less than 2 elements on the stack 
+* Fail if the stacks top 2 elements are not exactly 8 bytes 
+* Push the boolean result of the comparison onto the stack
+
+OP_NEG64
+* Fail if less than 1 element on the stack 
+* Fail if the stacks top is not exactly 8 bytes 
+* If the operation results in an overflow, push false onto the stack 
+* Push the result of negating the stack top onto the stack and push true onto the stack
+
+==Motivation==
+
+64 bit arithmetic operations are required to support arithmetic on satoshi values.
+Math on satoshis required precision of 51 bits. Many bitcoin protocol proposals - such as covenants -
+require Script access to output values. To support the full range of possible output values
+we need 64 bit precision.
```

Comment:
We can also add a paragraph about the disabled opcodes (MUL, DIV) and why it is considered safe to re-enable them in 64-bit form.